src/node/ledger_secrets.h

```diff
@@ -3,118 +3,283 @@
 #pragma once
 
 #include "crypto/symmetric_key.h"
-#include "ds/logger.h"
 #include "kv/kv_types.h"
+#include "kv/tx.h"
+#include "secrets.h"
 #include "tls/entropy.h"
 
 #include <algorithm>
-#include <nlohmann/json.hpp>
+#include <map>
+#include <optional>
 
 namespace ccf
 {
   struct LedgerSecret
   {
-    static constexpr auto MASTER_KEY_SIZE = crypto::GCM_SIZE_KEY;
-
-    std::vector<uint8_t> master; // Referred to as "sd" in TR
+    std::vector<uint8_t> raw_key;
+    std::shared_ptr<crypto::KeyAesGcm> key;
 
     bool operator==(const LedgerSecret& other) const
     {
-      return master == other.master;
+      return raw_key == other.raw_key;
     }
 
-    LedgerSecret()
+    LedgerSecret() = default;
+
+    LedgerSecret(const LedgerSecret& other) :
+      raw_key(other.raw_key),
+      key(std::make_shared<crypto::KeyAesGcm>(other.raw_key))
+    {}
+
+    LedgerSecret(std::vector<uint8_t>&& raw_key_) :
+      raw_key(raw_key_),
+      key(std::make_shared<crypto::KeyAesGcm>(std::move(raw_key_)))
+    {}
```

Comment:
Why do we call `make_shared` here - can't we copy-construct/move the original `key` directly?

```suggestion
    LedgerSecret(const LedgerSecret& other) :
      raw_key(other.raw_key),
      key(other.key)
    {}

    LedgerSecret(std::vector<uint8_t>&& raw_key_) :
      raw_key(raw_key_),
      key(std::move(other.key))
    {}
```

Reply:
For the copy constructor, yes. The second one isn't the move constructor so we still need the `make_shared` to create the key.